### PR TITLE
[circle-eval-diff] Introduce Tensor data structure

### PR DESCRIPTION
**Pull request by jinevening**
This introduces data structure for input/output tensor.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/8634
Draft PR: https://github.com/Samsung/ONE/pull/8741

### CODE REVIEW

File: compiler/circle-eval-diff/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,25 @@
 set(DRIVER "driver/Driver.cpp")
 
 file(GLOB_RECURSE SOURCES "src/*.cpp")
+file(GLOB_RECURSE TESTS "src/*.test.cpp")
+list(REMOVE_ITEM SOURCES ${TESTS})
+set(TEST_SOURCES
```
Comment (seanshpark) [inline compiler/circle-eval-diff/CMakeLists.txt:6]: this line can be moved after `ndif(NOT ENABLE_TEST)` line.

File: compiler/circle-eval-diff/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
 target_link_libraries(circle-eval-diff safemain)
+target_link_libraries(circle-eval-diff loco)
 target_link_libraries(circle-eval-diff vconone)
+
+if(NOT ENABLE_TEST)
+  return()
+endif(NOT ENABLE_TEST)
+
+nnas_find_package(GTest REQUIRED)
+GTest_AddTest(circle_eval_diff_test ${TESTS} ${TEST_SOURCES})
```
Comment (seanshpark) [inline compiler/circle-eval-diff/CMakeLists.txt:22]: Aboud adding `${TEST_SOURCES}`
- `circle-eval-diff` is executable so adding link to this exec looks strange
- so adding the source through `${TEST_SOURCES}` looks unavoidable.

Bu we can think this as optional;
- create `add_library(eval_diff_helper STATIC ${SOURCES})`
- add `target_link_libraries(circle-eval-diff eval_diff_helper)`
- and add `target_link_libraries(circle_eval_diff_test eval_diff_helper)` too

Comment (jinevening) [reply compiler/circle-eval-diff/CMakeLists.txt:22]: I've thought about that option, but I decided not to choose it.

`{SOURCES}` have all cpp files, which may require arser, safemain, vconone, which are not used by `circle_eval_diff_test`. To remove those dependencies, we'll have to select source files, similarly to the current way.

Comment (seanshpark) [reply compiler/circle-eval-diff/CMakeLists.txt:23]: Ah, OK then can you add some comments of this background facts?

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by llFreetimell — APPROVED** [review]
LGTM

**Pull request merged**